test/LocalVocabTest.cpp

```diff
@@ -4,19 +4,39 @@
 
 #include <gtest/gtest.h>
 
+#include <sstream>
+#include <string>
+
 #include "engine/ResultTable.h"
 #include "global/Id.h"
 
 TEST(LocalVocab, construction) {
-  auto makeIndex = [](size_t i) { return LocalVocabIndex::make(i); };
+  // Generate a test collection of words. For the tests below to work, these
+  // must all be different.
+  std::vector<std::string> testVocab(10'000);
+  for (size_t i = 0; i < testVocab.size(); ++i) {
+    testVocab[i] = std::to_string(i * 7635475567ULL);
+  }
+  // Create empty local vocabulary.
   LocalVocab localVocab;
   ASSERT_TRUE(localVocab.empty());
-  ASSERT_EQ(localVocab.getIndexAndAddIfNotContained("bla"), makeIndex(0));
-  ASSERT_EQ(localVocab.getIndexAndAddIfNotContained("blu"), makeIndex(1));
-  ASSERT_EQ(localVocab.getIndexAndAddIfNotContained("bli"), makeIndex(2));
-  ASSERT_EQ(localVocab.getIndexAndAddIfNotContained("bla"), makeIndex(0));
-  ASSERT_EQ(localVocab.size(), 3);
-  ASSERT_EQ(localVocab.getWord(LocalVocabIndex::make(0)), "bla");
-  ASSERT_EQ(localVocab.getWord(LocalVocabIndex::make(1)), "blu");
-  ASSERT_EQ(localVocab.getWord(LocalVocabIndex::make(2)), "bli");
+  // Add the words from our test vocabulary and check that they get the expected
+  // local IDs.
+  for (size_t i = 0; i < testVocab.size(); ++i) {
+    ASSERT_EQ(localVocab.getIndexAndAddIfNotContained(testVocab[i]),
+              LocalVocabIndex::make(i));
+  }
+  ASSERT_EQ(localVocab.size(), testVocab.size());
+  // Check that we get the same IDs if we do this again, but that no new words
+  // will be added.
+  for (size_t i = 0; i < testVocab.size(); ++i) {
+    ASSERT_EQ(localVocab.getIndexAndAddIfNotContained(testVocab[i]),
+              LocalVocabIndex::make(i));
+  }
+  ASSERT_EQ(localVocab.size(), testVocab.size());
+  // Check that the lookup by ID gives the correct words.
+  for (size_t i = 0; i < testVocab.size(); ++i) {
+    ASSERT_EQ(localVocab.getWord(LocalVocabIndex::make(i)), testVocab[i]);
+  }
+  ASSERT_EQ(localVocab.size(), testVocab.size());
```

Comment:
Great:)
I think this would cover a lot of possible errors now and will increase
stability over time for QLever!